validator/validator-main.protoascii

```diff
@@ -1771,37 +1773,164 @@ tags: {
 # the exception of tags inside of a <noscript> or <amp-story-player> ancestor
 # and transformed amp-img.
 # 4.7.1 The img element
+attr_lists: {
+  name: "img-attrs"
+  attrs: {
+    name: "align"
+    value_casei: "top"
+    value_casei: "middle"
+    value_casei: "bottom"
+    value_casei: "left"
+    value_casei: "right"
+  }
+  attrs: { name: "alt" }
+  attrs: { name: "border" }
+  attrs: {
+    name: "crossorigin"
+    value_casei: "anonymous"
+    value_casei: "use-credentials"
+  }
+  attrs: { name: "height" }
+  attrs: { name: "hspace" }
+  # https://developers.google.com/web/updates/2019/02/priority-hints
+  attrs: {
+    name: "importance"
+    value_casei: "high"
+    value_casei: "low"
+    value_casei: "auto"
+  }
+  attrs: { name: "ismap" }
+  attrs: {
+    name: "loading"
+    value_casei: "lazy"
+  }  # Not yet part of the html spec
+  attrs: { name: "name" }
+  attrs: {
+    name: "referrerpolicy"
+    value_casei: "no-referrer"
+    value_casei: "no-referrer-when-downgrade"
+    value_casei: "origin"
+    value_casei: "origin-when-cross-origin"
+    value_casei: "same-origin"
+    value_casei: "strict-origin"
+    value_casei: "strict-origin-when-cross-origin"
+    value_casei: "unsafe-url"
+  }
+  attrs: { name: "usemap" }
+  attrs: { name: "vspace" }
+  attrs: { name: "width" }
+}
+tags: {
+  html_format: AMP
+  tag_name: "IMG"
+  spec_name: "Standard Img"
+  descriptive_name: "img"
+  disallowed_ancestor: "AMP-IMG"
+  attrs: {
+    name: "decoding"
+    value_casei: "async"
+  }
+  attr_lists: "img-attrs"
+  attr_lists: "mandatory-src-or-srcset"
+}
+tags: {
+  html_format: AMP
+  tag_name: "IMG"
+  spec_name: "Hero Img"
+  descriptive_name: "img"
+  disallowed_ancestor: "AMP-IMG"
+  attrs: {
+    name: "data-hero"
+    mandatory: true
+    dispatch_key: NAME_DISPATCH
+  }
+  attrs: {
+    name: "decoding"
+    value_casei: "async"
+  }
+  attr_lists: "img-attrs"
+  attr_lists: "mandatory-src-or-srcset"
+}
+tags: {
+  html_format: AMP
+  tag_name: "IMG"
+  spec_name: "Img using srcset"
+  descriptive_name: "img"
+  disallowed_ancestor: "AMP-IMG"
+  attrs: {
+    name: "decoding"
+    value_casei: "async"
+  }
+  attrs: {
+    name: "sizes"
+  }
+  attr_lists: "img-attrs"
+  attr_lists: "mandatory-src-or-srcset"
+}
```

Comment:
Question: Why is there a separate `Img using srcset` spec? Since both `Standard Img` and `Hero Img` allow `srcset`, shouldn't `sizes` just be defined on those two specs above?

Reply:
@westonruter Thank you for your feedback.  Is there a spec that indicates this? I couldn't find a definitive doc that indicates this that's why I fell back on the assumption that srcset is not allowed for hero images.  If it is, I can change the validator to reflect this.

Reply:
@banaag Maybe this was confused by the fact that image preloading doesn't work for responsive images except by Chromium? Only Chromium supports `imagesrcset` and `imagesizes` on `link[rel=preload][as=image]`: https://web.dev/preload-responsive-images/#responsive-images-+-preload-faster-image-loads

But this shouldn't impact whether or not an author can have a hero image with `srcset`/`sizes`. The only difference between a hero image and a non-hero image should be whether or not the `loading` attribute is present. Hero images should never have `loading=lazy` because this causes a slight delay for loading and hurts LCP.

If a hero image cannot have `srcset`/`sizes` and thus cannot be a responsive image, then such images will render poorly for some visitors if they can't have an appropriately-sized image served.

The AMP Optimizers will automatically generate a `link[rel=preload]` for hero images that lack `srcset`/`sizes`, but they currently require an opt-in to do so for responsive hero images (since not all browsers yet support `imagesrcset`/`imagesizes`.

So yes, hero images should allow `srcset`/`sizes`.

Reply:
@westonruter Thanks for the clarification.  Will send out a follow-on PR to have hero images support srcset/sizes.